### PR TITLE
Extend HMI:JsonTable, create Edit CSV POU, create example, add some doc

### PR DESCRIPTION
**Pull request by kraskrom**

### CODE REVIEW

File: svghmi/widget_image.ysl2 [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    ||
+
+    shortdesc > Image display
+}
+
+widget_class("Image")
+    ||
+        frequency = 5;
+        dispatch(value, oldval, index) {
+            this.fields[index] = value;
```
Comment (etisserant) [inline svghmi/widget_image.ysl2]: "fields" array is not needed here. Store only if index == 0.
```suggestion
            if(index==0)
                this.given_url = value;
```
Same for "readyfields", only "ready" member should be needed

File: py_ext/py_ext.py [hunk 2]
```diff
@@ -83,24 +82,28 @@ def CSVRdStr(fname, rowname, colname):
...
-            csvfile = open(fname, 'rb')
+            csvfile = open(fname, 'rt', encoding='utf-8')
         except IOError:
             return "#FILE_NOT_FOUND"
         try:
             dialect = csv.Sniffer().sniff(csvfile.read(1024))
             csvfile.seek(0)
             reader = csv.reader(csvfile, dialect)
-            headers = dict([(name, index) for index, name in enumerate(reader.next()[1:])])
+            first_row = True
```
Comment (etisserant) [inline py_ext/py_ext.py]: Why is this change ? Isn't csv.reader an iterator ?

File: svghmi/widget_jsontable.ysl2 [hunk 3]
```diff
@@ -189,8 +189,13 @@ template "svg:text", mode="json_table_elt_render" {
     }
 }
 
+template "svg:image", mode="json_table_elt_render" {
+    param "expressions";
+    const "value_expr", "$expressions/expression[1]/@content";
+    |         id("«@id»").setAttribute('xlink:href', String(«$value_expr»));
```
Comment (etisserant) [inline svghmi/widget_jsontable.ysl2]: "xlink:href" is deprecated. This should be accessible as an object member with ".href"

File: py_ext/py_ext.py (hunk 2 above)
Comment (kraskrom) [reply py_ext/py_ext.py]: The function `next()` no longer exists, so I initially just used the loop through rows. I've now replaced this with `__next__()` function

**Comment by etisserant** [comment]
All good for me. Many thanks !

**Pull request merged**